### PR TITLE
Common Library for Handshaking

### PR DESCRIPTION
**Pull request by sonndinh**

### CODE REVIEW

File: tactical-microgrid-standard/common/Handshaking.mpc [hunk 1]
```diff
@@ -0,0 +1,12 @@
+project: opendds_cxx11 {
+
+  includes += ../idl
+  libpaths += ../idl
+  libs += tms
```
Comment (iguessthislldo) [inline tactical-microgrid-standard/common/Handshaking.mpc:5]:
```suggestion
  libs += tms
  after += tms
```

File: tactical-microgrid-standard/common/Handshaking.cpp [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+}
+
+DDS::ReturnCode_t Handshaking::start_heartbeats(tms::Identity id)
+{
+  if (!hb_dw_) {
+    ACE_ERROR((LM_NOTICE, "(%P|%t) NOTICE: Handshaking::start_heartbeats: create data writers first with create_publishers!\n"));
+    return DDS::RETCODE_ERROR;
+  }
+
+  ACE_Time_Value period(1, 0);
```
Comment (iguessthislldo) [inline tactical-microgrid-standard/common/Handshaking.cpp:151]:
```suggestion
  const ACE_Time_Value period(1, 0);
```

I'm honestly surprised this isn't a constant in the IDL.

File: tactical-microgrid-standard/common/HeartbeatDataReaderListenerImpl.h [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    : callback_(cb)
+  {}
+
+  virtual ~HeartbeatDataReaderListenerImpl() {}
+
+  virtual void on_requested_deadline_missed(DDS::DataReader_ptr,
+                                            const DDS::RequestedDeadlineMissedStatus&)
+  {
+    ACE_DEBUG((LM_INFO, "(%P|%t) INFO: HeartbeatDataReaderListenerImpl::on_requested_deadline_missed\n"));
+  }
```
Comment (iguessthislldo) [inline tactical-microgrid-standard/common/HeartbeatDataReaderListenerImpl.h:22]: This is just a suggestion, but if there's going to be more listeners where we're not going to use most of the methods, maybe use a common base sorta like the one I started to work on in https://github.com/OpenDDS/OpenDDS/pull/3460.

Comment (sonndinh) [reply tactical-microgrid-standard/common/HeartbeatDataReaderListenerImpl.h:22]: I agree. I think there'll be more listeners when more topics are added.

**Pull request merged**